contracts/schema/mutations.py

```diff
@@ -0,0 +1,40 @@
+import graphene
+from django.utils.translation import gettext_lazy as _
+
+import contracts
+from berth_reservations.exceptions import VenepaikkaGraphQLError
+from payments.models import Order
+
+
+class FulfillContractMutation(graphene.ClientIDMutation):
+    class Input:
+        order_number = graphene.NonNull(graphene.String)
+        return_url = graphene.NonNull(graphene.String)
+        identifier = graphene.NonNull(graphene.String)
+        auth_service = graphene.NonNull(graphene.String)
+
+    signing_url = graphene.String()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **input):
+        try:
+            order = Order.objects.get(order_number=input.get("order_number"))
+        except Order.DoesNotExist:
+            raise VenepaikkaGraphQLError(_("No order found for given order number"))
+
+        try:
+            contract = order.lease.contract
+        except Exception:
+            raise VenepaikkaGraphQLError(_("No contract found for given order"))
+
+        signing_url = contracts.services.get_contract_service().fulfill_contract(
```

Comment:
This was how it was coded at first, but the named import messes with the global test monkeypatch. In `berth_reservations/tests/conftest.py` `get_contract_service` is mocked to always return an instance of `TestContractService`. This is to prevent someone accidentally writing a test case that would hit the actual visma backend.

If there's a better way to go about this I'll gladly change it.